homeassistant/components/ness_alarm/alarm_control_panel.py

```diff
@@ -41,7 +41,7 @@ async def async_setup_platform(
     if discovery_info is None:
         return
 
-    device = NessAlarmPanel(hass.data[DATA_NESS], "Alarm Panel")
+    device = NessAlarmPanel(hass.data[DATA_NESS])
```

Comment:
It's been a while since I've developed for core, but I think if we used `async_setup_entry` (as per config flow), to create both the client + the entity together, allowing you to pass the name directly to the entity instance instead of via the client instance.